### DISCUSSION TITLE
AniList metrics "An error occurred"

### DISCUSSION THREAD

**Discussion by stauersbol** [lowlighter/metrics#1019] [Q&A] [answered]
I've recently begun to get this issue with AniList metrics, I looked for any possible updates, but nothing changed after changing the uses from `lowlighter/metrics@master` to `lowlighter/metrics@latest`.

Is there any coming fix to this?

![metrics plugin personal anilist](https://user-images.githubusercontent.com/9026132/165278858-b389b675-a063-4f0c-bdcc-0d8879e585cd.svg)

**Comment by lowlighter** [comment, marked as answer]
Should be fixed by #1020

Not sure which anime/manga you watched recently, but it seems that one of them add a `null` description which was causing issues with the linebreak sanitizer

**Reply by stauersbol** [reply]
Weird. I tried running it after you pushed that fix, but still getting the same error.

**Reply by lowlighter** [reply]
Did you use `@master` when you tried again 🤔 ?
If yes, can you set `debug: yes` to enable logs ?

**Reply by stauersbol** [reply]
No I didn't use `@master`. Still used `@latest` for that. I could try with master and set `debug: yes`

**Reply by stauersbol** [reply]
I enabled `debug: yes` on `@latest`. I checked for errors and found this:

```
{
    name: 'anilist',
    result: {
      error: {
        message: 'An error occured',
        instance: TypeError: Cannot read properties of null (reading 'replace')
            at format (file:///metrics/source/plugins/anilist/index.mjs:146:30)
            at file:///metrics/source/plugins/anilist/index.mjs:41:67
            at Array.map (<anonymous>)
            at Object.default [as anilist] (file:///metrics/source/plugins/anilist/index.mjs:41:54)
            at async file:///metrics/source/plugins/core/index.mjs:63:30
            at async Promise.all (index 0)
            at async metrics (file:///metrics/source/app/metrics/index.mjs:64:22)
            at async retry.retries.retries (file:///metrics/source/app/action/index.mjs:395:26)
            at async retry (file:///metrics/source/app/action/index.mjs:56:22)
            at async file:///metrics/source/app/action/index.mjs:394:20
      }
    }
  }
```

**Reply by stauersbol** [reply]
Besides that I also got this, it is a pastebin link: https://pastebin.com/mFdP9qr8
Error message is too long to paste it here directly.

**Reply by lowlighter** [reply]
The fix is currently on @master only, @latest branch is updated on each release so the fix isn't on it for now.

Let me know if you stil have this issue if you switch on master branch

Concerning the second error, it's a non-fatal one, it's just that it tries to fetch emoji list but when token is set to `NOT_NEEDED` it fails as since GitHub refuse the request. Should probably hide it, but it doesn't really impact the workflow though

**Reply by stauersbol** [reply]
Error is solved on master branch now. Only thing was the last message I mentioned. But that - as you said - has no impact on it being successful.
